draft-ietf-dnsop-svcb-https.md

```diff
@@ -1082,18 +1082,31 @@ single-client granularity.
 
 ## Interaction with Alt-Svc
 
-Clients that do not implement support for Encrypted ClientHello MAY
-skip the HTTPS RR query
-if a usable Alt-Svc value is available in the local cache.
-If Alt-Svc connection fails, these clients SHOULD fall back to the HTTPS RR
-client connection procedure ({{client-behavior}}).
-
-Clients that implement support for ECH MUST perform the HTTPS RR query first,
-and MUST only make use of Alt-Svc when operating in SVCB-optional mode (see
-{{ech-client-behavior}}).
-
-This specification does not alter the DNS records used when connecting
-to an Alt-Svc hostname (typically A and/or AAAA only).
+Clients that implement support for both Alt-Svc and HTTPS records SHOULD
+retrieve any HTTPS records for the Alt-Svc alt-authority, and ensure that
+their connection attempts are consistent with both the Alt-Svc parameters
+and any received HTTPS SvcParams.  If present, the HTTPS record's target
+and port override the alt-authority.  For example, suppose
+"https://example.com" sends an Alt-Svc field value of:
+
+    Alt-Svc: h3="alt.example.com:8443"; ...
+
+The client would retrieve the following HTTPS record:
+
+    _8443._https.alt.example.com. ... 1 alt2.example.com. (
+        port=9443 alpn=h2,h3 ech=... )
+
+The client could then attempt an HTTP/3 connection to `alt2.example.com:9443`
+with ECH, as this is consistent with both the Alt-Svc field value and the
+HTTPS record.  This specification does not alter or clarify the
+interpretation of Alt-Svc's `protocol-id` field.
```

Comment:
Do we also want to add a:
"Clients MAY choose to ignore individual returned HTTPS RRs,
or the Alt-Svc entry entirely, based on mismatches between the 
Alt-Svc record and HTTPS RR SvcParams."
This gives flexibility here for client behavior based on ECS
as well as for cases like the Chrome behavior of wanting
to match h3-to-h3 and h2-to-tcp but without over-specifying.

Reply:
I'm not sure I understand, but I don't think this is necessary.  If a record cannot be used in a way that is consistent with the Alt-Svc Field Value, then the record is effectively ignored when attempting to use that Alt-Svc Field Value.

Reply:
This would be giving the client permission to do filtering.  In the above example, h3 would only be used for the HTTPS RR at _8443._https.alt.example.com and h2 would only be used for the HTTPS RR at alt.example.com

Reply:
In this example, neither RR is "ignored".  They're "constrained" or "filtered", perhaps.

I've expanded the text to match the new example.  I think the new text conveys the idea pretty clearly.